src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *     import org.openapitools.client.model.CreateIndexRequestSpecPodMetadataConfig;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     CreateIndexRequestSpecPodMetadataConfig metadataConfig = new CreateIndexRequestSpecPodMetadataConfig().fields(Arrays.asList("genre", "year"));
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", metadataConfig);
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param metric the metric type for the index. Must be one of "cosine", "euclidean", or "dot_product".
+     * @param metadataConfig the configuration for the behavior of Pinecone's internal metadata index. By default, all metadata is indexed;
+     *                       when metadataConfig is present, only specified metadata fields are indexed.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, String metric, CreateIndexRequestSpecPodMetadataConfig metadataConfig) {
-        validatePodIndexParams(indexName, dimension, environment, podType, metric, null, null, null);
-
-
         return createPodsIndex(indexName, dimension, environment, podType, metric, null, null, null, metadataConfig,
                 null);
     }
 
-    // Minimal + source collection
+    /**
+     * Overload for creating a new pod index with environment, podType, and metric, and sourceCollection.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", "my-collection");
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param metric the metric type for the index. Must be one of "cosine", "euclidean", or "dot_product".
+     * @param sourceCollection the name of the collection to be used as the source for the index.
```

Comment:
Not mandatory, but it might help clear up confusion to state that collections are a static snapshot of a single index at a particular point in time. that way it's clear that it's sorta like creating an index from another index (to me, the term `collection` sounds like it can contain 1+ indexes, which it cannot)